### ISSUE TITLE
Continuations not working on live

### ISSUE THREAD

**Issue by Steven-Eardley** [DOAJ/doaj#696]
TOCs for journal continuations are being redirected to the most recent version - i.e. the previous title link has no effect, and OpenURL queries return the wrong TOC. Additionally, it redirects twice; with ```http``` then ```https```.

Might be worth going over changes made in PR #678 again, @emanuil-tolev? Could be something more recent, though.

* https://doaj.org/openurl?url_ver=Z39.88-2004&url_ctx_fmt=info:ofi/fmt:kev:mtx:ctx&rft_val_fmt=info:ofi/fmt:kev:mtx:journal&rft.genre=journal&rft.issn=0366-1644
* https://doaj.org/toc/0366-1644

**Comment by emanuil-tolev** [comment]
sort of unrelated note: why doesn't this cause a whitespace compile error? https://github.com/DOAJ/doaj/pull/678/files#diff-15fcdea4c39a69a3a91220826502d799R165

(just wondering if it WILL cause an error... should fix that together with whatever the fix is for this issue)

**Comment by emanuil-tolev** [comment]
> https://doaj.org/toc/0366-1644

OK so for this: that ISSN is in the index.issn part of the record, which is how the find_by_issn method works. This is the record it's finding: https://doajgate.cottagelabs.com/doaj/journal/30a6fbcbc2a241ffbb376da7be41b29c

Afterwards it simply redirects to the current E-ISSN of the journal. It assumes the ISSN in the request is assigned to the *current* journal record (because it found that with the find_by_issn method...), and redirects you to the E-ISSN of the current journal record (which is what it's supposed to do).

So, not sure why that 0366-1644 ISSN is recorded in the current record. Should it be? Do we want to find records by their past issns when calling find_by_issn? It might be easiest for me to talk this over with @richard-jones , perhaps at the Fri meeting..

> Additionally, it redirects twice; with http then https.

Not sure why the first redirect is happening, it should just direct you to the HTTPS page. The 2nd redirect is just because we enforce HTTPS everywhere (except the doajArticles XSD schema, apparently), that's the right behaviour there.

**Comment by Steven-Eardley** [comment]
> https://doaj.org/toc/0366-1644
OK so for this: that ISSN is in the index.issn part of the record, which is how the find_by_issn method works. This is the record it's finding: https://doajgate.cottagelabs.com/doaj/journal/30a6fbcbc2a241ffbb376da7be41b29c

> Afterwards it simply redirects to the current E-ISSN of the journal.

You'll notice that's the url in the 'Previous title(s):' link on the page - is there a new way to link to the TOC for a continuation?

**Comment by emanuil-tolev** [comment]
> You'll notice that's the url in the 'Previous title(s):' link on the page - is there a new way to link to the TOC for a continuation?

No, well, yes - you're doing it right, but it's not working, because the current (not past) record has the old ISSN. Not sure what to do (delete the ISSN from index.ISSN in that record? change the code?) hence the suggestion to talk on Fri.

**Comment by emanuil-tolev** [comment]
> So, not sure why that 0366-1644 ISSN is recorded in the current record. Should it be?

@richard-jones says Yes.

> Do we want to find records by their past issns when calling find_by_issn?

Yes, that's how continuations work.

I need to move my redirection code below the code that considers continuations.

**Comment by Steven-Eardley** [comment]
The fix is live now. Solved the problem I originally reported - leaving for @dommitchell to determine if TOCs on continuations are as expected.

**Comment by dommitchell** [comment]
Is there an update on this one?

**Comment by emanuil-tolev** [comment]
The problem here is that the continuations implementation overrides which bibjson is being shown (the historic bibjson) but the journal object remains as the current journal. This is because at the time of writing PVOD, continuations were a bunch of data related to a current journal and not treated as a first-class journal object. The continuations & workflow mgmt project should make this better.

For the time being I've stopped the ToC page from relying on the current journal object, and instead directed its attention at the correct place (the data which could be current, or could be continuations' metadata). It works now, on my machine. I've submitted a PR and this will be fixed shortly.
